packages/runtime-vapor/src/component.ts

```diff
@@ -144,8 +144,9 @@ export function createComponent(
     locateHydrationNode()
   }
 
+  const isFnComponent = isFunction(component)
   // vdom interop enabled and component is not an explicit vapor component
-  if (appContext.vapor && !component.__vapor) {
+  if (appContext.vapor && !isFnComponent && !component.__vapor) {
```

Comment:
Need to distinguish whether a functional component is a vdom component or a vapor component